cli/src/main/java/com/box/l10n/mojito/cli/command/ExtractionCheckCommand.java

```diff
@@ -368,6 +383,51 @@ private void generateSarifFile(
     }
 
     Sarif sarif = sarifFileGenerator.generateSarifFile(cliCheckerFailures, assetExtractionDiffs);
+
+    if (shouldValidateSarifOutput && githubClients.isClientAvailable(githubOwner)) {
+      logger.debug("Validating SARIF output");
+      Map<String, Set<Integer>> githubFileToLineNumberMap =
+          githubClients
+              .getClient(githubOwner)
+              .getPrFilePatches(githubRepository, githubPRNumber)
+              .entrySet()
+              .stream()
+              .collect(
+                  Collectors.toMap(
+                      Map.Entry::getKey,
+                      entry -> githubPatchParser.getAddedLines(entry.getValue())));
+
+      logger.debug("GitHub file to line number map: {}", githubFileToLineNumberMap);
+      Map<String, Set<Integer>> sarifFileToLineNumberMap =
+          SarifUtils.buildFileToLineNumberMap(sarif);
+      logger.debug("Sarif file to line number map: {}", sarifFileToLineNumberMap);
+      Map<String, Set<Integer>> misreportedLineNumbersPerFile =
+          getMismatchedFileWithLineNumbers(sarifFileToLineNumberMap, githubFileToLineNumberMap);
+      logger.debug("MisreportedLineNumbersPerFile size: {}", misreportedLineNumbersPerFile.size());
+      logger.debug(
+          "MisreportedLineNumbersTotal: {}",
+          misreportedLineNumbersPerFile.values().stream().mapToInt(Set::size).sum());
```

Comment:
I think we should also output some of these stats to the console if they're greater than 1, should make for easier debugging

Reply:
Fair